lib/vanagon/platform/rpm.rb

```diff
@@ -48,10 +48,29 @@ def rpm_defines
         defines =  %(--define '_topdir $(tempdir)/rpmbuild' )
         # RPM doesn't allow dashes in the os_name. This was added to
         # convert cisco-wrlinux to cisco_wrlinux
-        unless is_aix?
-          defines << %(--define 'dist .#{@os_name.gsub('-', '_')}#{@os_version}' )
+        defines << %(--define 'dist .#{@os_name.gsub('-', '_')}#{@os_version}' )
+      end
+
+      def add_repository(definition)
+        definition = URI.parse(definition)
+
+        commands = ["rpm -q curl > /dev/null || yum -y install curl"]
+        if definition.scheme =~ /^(http|ftp)/
+          if File.extname(definition.path) == '.rpm'
+            # repo definition is an rpm (like puppetlabs-release)
+            commands << "curl -o local.rpm '#{definition}'; rpm -Uvh local.rpm; rm -f local.rpm"
+          else
```

Comment:
should we also be installing a gpg key in this case, like we are for deb? It's probably not something for this PR, but something to think about